drivers/dma/dma_nxp_edma.c

```diff
@@ -583,19 +578,36 @@ static int edma_get_attribute(const struct device *dev, uint32_t type, uint32_t
 
 static bool edma_channel_filter(const struct device *dev, int chan_id, void *param)
 {
-	int *requested_channel;
+	struct edma_channel *chan;
 
 	if (!param) {
 		return false;
 	}
 
-	requested_channel = param;
+	if (*(int *)param != chan_id) {
+		return false;
+	}
 
-	if (*requested_channel == chan_id && lookup_channel(dev, chan_id)) {
-		return true;
+	chan = lookup_channel(dev, chan_id);
+	if (!chan) {
+		return false;
 	}
 
-	return false;
+	irq_enable(chan->irq);
```

Comment:
You add the `irq_enable()` on `dma_chan_filter()` - which is a different function than `dma_request_channel(`) as mentioned in the commit message.

Reply:
ah, the commit message is supposed to refer to `dma_request_channel` and `dma_release_channel`. The filter function is called as part of `dma_request_channel`.